packages/react-form-state/src/validators.ts

```diff
@@ -137,16 +145,40 @@ const validators = {
     return validate(isNumericString, errorContent);
   },
 
-  requiredString(errorContent: ErrorContent) {
-    return validate(not(isEmptyString), errorContent);
-  },
-
   nonNumericString(errorContent: ErrorContent) {
     return validate(not(isNumericString), errorContent);
   },
 
+  requiredString(errorContent: ErrorContent) {
+    return (input: string) => {
+      if (not(isEmptyString)(input)) {
+        return;
+      }
+
+      if (typeof errorContent === 'function') {
+        // eslint-disable-next-line consistent-return
+        return errorContent(toString(input));
+      }
+
+      // eslint-disable-next-line consistent-return
+      return errorContent;
+    };
+  },
+
   required(errorContent: ErrorContent) {
-    return validate(not(isEmpty), errorContent);
+    return (input: any) => {
```

Comment:
Wondering if `any` should be `string` here. An input field will always be a string, no?

Reply:
possibly makes sense, I was defaulting to `input` `any` to let validators be used even on object / array fields, but maybe that doesn't really make sense